src/plugins/mer/meremulatordevice.cpp

```diff
@@ -807,43 +809,76 @@ void MerEmulatorDeviceManager::onDeviceListReplaced()
 
         progress.setLabelText(tr("Applying virtual machine settings: '%1'").arg(merEmulator->virtualMachine()));
 
-        const quint16 nowSshPort = merEmulator->sshParameters().port();
+        quint16 nowSshPort = merEmulator->sshParameters().port();
         if (nowSshPort != oldSshPortsCache.value(merEmulator->id())) {
-            MerVirtualBoxManager::updateEmulatorSshPort(merEmulator->virtualMachine(), nowSshPort);
-            merEmulator->updateConnection();
+            if (MerVirtualBoxManager::updateEmulatorSshPort(merEmulator->virtualMachine(), nowSshPort)) {
+                merEmulator->updateConnection();
+            } else {
+                nowSshPort = oldSshPortsCache.value(merEmulator->id());
+                QSsh::SshConnectionParameters nowShhParameters = merEmulator->sshParameters();
+                nowShhParameters.setPort(nowSshPort);
+                merEmulator.constCast<MerEmulatorDevice>()->setSshParameters(nowShhParameters);
+                ok = false;
+            }
         }
         m_deviceSshPortCache.insert(merEmulator->id(), nowSshPort);
 
-        const Utils::PortList nowQmlLivePorts = merEmulator->qmlLivePorts();
-        if (nowQmlLivePorts.toString() != oldQmlLivePortsCache.value(merEmulator->id()).toString())
-            MerVirtualBoxManager::updateEmulatorQmlLivePorts(merEmulator->virtualMachine(),
+        Utils::PortList nowQmlLivePorts = merEmulator->qmlLivePorts();
+        if (nowQmlLivePorts.toString() != oldQmlLivePortsCache.value(merEmulator->id()).toString()) {
+            Utils::PortList savedPorts = MerVirtualBoxManager::updateEmulatorQmlLivePorts(merEmulator->virtualMachine(),
                     merEmulator->qmlLivePortsList());
-        m_deviceQmlLivePortsCache.insert(merEmulator->id(), nowQmlLivePorts);
 
-        const int nowMemorySize = merEmulator->memorySizeMb();
-        if (nowMemorySize != oldMemorySizeCache.value(merEmulator->id()))
-            MerVirtualBoxManager::setMemorySizeMb(merEmulator->virtualMachine(), nowMemorySize);
+            if (savedPorts.toString() != nowQmlLivePorts.toString()) {
```

Comment:
Add lambda to check PortList. Yes, its work as you say

Reply:
OK, not ideal (on failure it will replace ranges with plain enumerations) but let it be like this :) Just please remove the extra braces around one of the `if` blocks